definitions/features/downstream.rb

```diff
@@ -11,8 +11,39 @@ def current_version
     @current_version ||= rpm_version('satellite') || version_from_source
   end
 
+  def current_minor_version
+    current_version.to_s[/^\d+\.\d+/]
+  end
+
+  def setup_repositories(version)
+    activation_key = ENV['EXTERNAL_SAT_ACTIVATION_KEY']
+    org = ENV['EXTERNAL_SAT_ORG']
+    if activation_key
+      org_options = org ? %(--org #{shellescape(org)}) : ''
+      execute!(%(subscription-manager register #{org_options}\
+                  --activationkey #{shellescape(activation_key)} --force))
+    else
+      execute!(%(subscription-manager repos --disable '*'))
+      enable_options = rh_repos(version).map { |r| "--enable=#{r}" }.join(' ')
+      execute!(%(subscription-manager repos #{enable_options}))
+    end
+  end
+
   private
 
+  def rh_repos(sat_version)
+    sat_version = version(sat_version)
+    ["rhel-#{rh_version.major}-server-rpms",
```

Comment:
Would it make sense to add verification of available repos as a pre-upgrade check to avoid failure during the "changing" phase? Should I file a RFE with this?